drivers/adc/adc_context.h

```diff
@@ -255,6 +255,7 @@ static inline void adc_context_on_sampling_done(struct adc_context *ctx,
 		switch (action) {
 		case ADC_ACTION_REPEAT:
 			repeat = true;
+			ctx->sampling_index = 0U;
```

Comment:
This is wrong. When the callback returns `ADC_ACTION_REPEAT`, it means that https://github.com/zephyrproject-rtos/zephyr/blob/d1591cd363d5cc23929fbd85ddbf79c4b106a377/include/zephyr/drivers/adc.h#L538-L541 so the sampling index should stay unchanged. The last sampling is to be repeated, not the entire sequence.

Reply:
Hmm, it depends on how you interpret this doxygen comment.

A) What would be the useful scenario where ACTION_REPEAT would be used in combination with extra_sampling > 0?
When using ACTION_REPEAT in combination with extra_sampling == 0, then it makes sense, taking another sample and write it into the same buffer location. Essentially a sequence of 1.

B) How would I then repeat an entire sequence, by manually starting ADC read_async again? That creates additional execution delay, essentially reprogramming ADC and DMA with identical values.